site/content/blog/2019-10-21-localizing-matterpoll/index.md

```diff
@@ -0,0 +1,182 @@
+
+The [Mattermost Server](https://github.com/mattermost/mattermost-server) uses the [go-i18n package](https://github.com/nicksnyder/go-i18n). The library is well maintained and very popular, which gave me confidence in using it.
+
+
+#### Choosing a version
+
+When [Translations where added to Mattermost](https://github.com/mattermost/mattermost-server/commit/8e404c1dcf820cf767e9d6899e8c1efc7bb5ca96#diff-db85c0ea4d2e69c8abaefa875ba77c51) back in 2016 the latest version of go-i18 available was [`v1.4.0`](https://github.com/nicksnyder/go-i18n/releases/tag/v1.4.0). Since then the development has continued and [`v2.0.0`](https://github.com/nicksnyder/go-i18n/releases/tag/v2.0.0) was released may this year. V2 has a completely different API and also the CLI tool [`goi18n`](https://github.com/nicksnyder/go-i18n#command-goi18n-) has changed a lot. Hence, we had to make a choice: Stick with the well established V1 or use the newer V2? 
+
+Let's dive into the difference between the to versions. In V1 translation string are define in the translation files, e.g. `en-US.all.json` would contain:
+```json
+  {
+    "id": "settings_title",
+    "translation": "Settings"
+  }
+```
+These translations would then be accessible in the go code with a translation function (`T("settings_title")`). This leads to a loose couping between the translations string in the translation file and in the code. Often developers forget to add a translation string or misspelled the id.
+
+V2 fixed this by defining the translations string in the code and automatically extracting it into the translation file. This greatly improves the experience for developers. Also, there are now **two** translation files for every languages. For e.g. german there is `active.de.json`, which contains already translated strings, and `translate.de.json`, which contains strings that have to be translated. They either are newly added strings or string where the text changed. `goi18n` helps with auto populating these files.
+
+This changes is very handy for both developers and contributors and the main reason we ended up choosing V2 for Matterpoll. Other notable changes are:
+
+- Better support for plural forms with options for e.g. `Zero`, `One`, `Few` and `Many`
+- No global state, which helps with running tests in parallel
+- Usage of [`golang.org/x/text/language`](https://godoc.org/golang.org/x/text/language) for standardized behavior
+
+For further information please take a look at the [Changelog](https://github.com/nicksnyder/go-i18n/blob/master/CHANGELOG.md#v2) of go-i18n.
+
+Also, it's a good idea to stay up to date with the major version of the libraries that you are using. This way you get the latest bug fixes because open libraries don't backport fixes.
+
+With all this in mind it was not a hard decision for us to go with V2 of go-i18n.
+
+
+#### Choosing a file format
```

Comment:
```suggestion
#### Choosing a File Format
```

Reply:
See https://github.com/mattermost/mattermost-developer-documentation/pull/421/files#r344621103

Reply:
Replied on that.